src/arm7tdmi.rs

```diff
@@ -1,9 +1,9 @@
-use std::fmt::Error;
+use std::{fmt::Error, rc::Rc};
 
 use crate::{condition::Condition, cpsr::Cpsr, cpu::Cpu};
 
 pub(crate) struct Arm7tdmi {
-    rom: Vec<u8>,
+    rom: Rc<Vec<u8>>,
```

Comment:
Not sure we need `Rc` right now, but open to proposals :)

Reply:
I agree it isn't needed, it was just to separate the ROM from the CPU itself and share the variable with the cartridge, a ref is also an idea. I don't think that the CPU itself should own memory stuff directly

Reply:
Personally I like it!

Reply:
Probably will be owned by something like `InternalMemory` or sort of, for now I think `&` is enough but let's keep this if other people like it :)